search.html

```diff
@@ -200,18 +189,37 @@
     async: false,
     success: function(data) {
       $.each( data, function( key, val ) {
-        if ((search_type === val.type || search_type === "all")
-          && (search_tags.length == 0 || $.inArray(val.tag ,search_tags) != -1 )) {
-          elasticlunr_index.addDoc(val);
-          store[val.id] = {
-            type: val.type,
-            title: val.title,
-            url: val.url,
-            page_url: val.page_url,
-            image: val.image,
-            content: val.content,
-            preview: val.preview
-          };
+        if ((search_type === val.type
```

Comment:
I don't understand why there are two different cases for adding documents to the index; in fact, I'm not even seeing what the difference is in the logic.  Do we really need both?  If so, needs documentation as to why.  But it would be better if we could consolidate them.